core/backend/src/CheckpointManager.ts

```diff
@@ -275,11 +275,15 @@ export class V1CheckpointManager {
         if (dbGuid !== Guid.normalize(requestedCkp.iModelId)) {
           Logger.logWarning(loggerCategory, "iModelId is not properly setup in the briefcase. Updated briefcase to the correct iModelId.", () => ({ ...traceInfo, ...checkpoint, dbGuid }));
           nativeDb.setDbGuid(Guid.normalize(requestedCkp.iModelId));
+          // Required to reset the ChangeSetId because setDbGuid clears the value.
+          nativeDb.saveLocalValue("ParentChangeSetId", dbChangeSetId);
         }
 
         const dbContextGuid = Guid.normalize(nativeDb.queryProjectGuid());
-        if (dbContextGuid !== Guid.normalize(requestedCkp.contextId))
-          throw new IModelError(IModelStatus.ValidationFailed, "ContextId was not properly setup in the briefcase", Logger.logError, loggerCategory, () => ({ ...traceInfo, dbContextGuid }));
+        if (dbContextGuid !== Guid.normalize(requestedCkp.contextId)) {
+          Logger.logWarning(loggerCategory, "ContextId was not properly setup in the briefcase. Updated briefcase to the correct ContextId.", () => ({ ...traceInfo, ...checkpoint, dbContextGuid }));

```

Comment:
a nit, but this should say checkpoint rather than briefcase.

Reply:
I'll update this in the follow-up test PR.

Reply:
Fixed in #624